WordPress/gutenberg
# Interactivity Router: Move ARIA live region and loading bar to the Interactivity Router

## What?

Epic: https://github.com/WordPress/gutenberg/issues/56803

Move the logic that renders and controls the loading top bar and the ARIA live region―both elements that inform about client-side navigations―from the Query block to the Interactivity Router module and the `wp-router-region` directive processor.

Also, the `@wordpress/interactivity-router` module exposes `state.url`, containing the current URL.

## Why?

We want that logic to be general for any block using client-side navigations, not just for the Query block.

## How?

- All the JS logic has been moved to the `@wordpress/interactivity-router` module.
- All the PHP logic has been moved to the private method `data_wp_router_region_processor`, inside the `WP_Interactivity_API` class.

## Testing Instructions
1. In the site editor, deactivate "Force page reload" for the Query block.
2. Visit the homepage.
3. In the browser dev tools, add a network throttling (e.g. Slow 3G).
4. In the Query block, navigate to the Next page.
5. Ensure the loading bar appears and start an animation.
6. Ensure the loading bar disappears when the next page is loaded.

### Testing Instructions for Keyboard
1. Do previous steps 1 to 3.
2. Enable a screen reader (e.g., VoiceOver).
4. In the Query block, navigate to the Next page.
5. Ensure the message "Loading page, please wait" is announced.
5. When the page has finished loading, the "Page loaded" message should be heard.
